src/test/java/com/google/javascript/cl2dts/shouldResolveForwardDeclares/forward.js

```diff
@@ -0,0 +1,11 @@
+goog.provide('forward.A');
```

Comment:
That makes sense. I added a single file test with `goog.forwardDeclare` in addition to what we had.

I was unfortunately using the term forwardDeclare in different situation, which I renamed to `namedTypes`. The situation is the following:

`a.js` and `b.js` are part of the compilation. Class `A` in `a.js` uses type `B` from `b.js` but doesn't goog.require it. Both files are passed to closure. Turns out that if the order that those files are passed matters for the internal representation of types. If `a.js` is seen first type `B` is what internally is called a `namedType`. Outputting the named type without resolving is problematic for generic types, which don't need to be concertized in closure, but do need to be in TS. 

The fix in this PR is to follow the resolved type for namedTypes instead of just outputting the name.